vowpalwabbit/slim/src/opts.cc

```diff
@@ -6,6 +6,52 @@
 namespace vw_slim
 {
 // && used to avoid constant string copying
+void find_opt(std::string const& command_line_args, std::string arg_name, std::vector<std::vector<uint8_t>> out_values)
```

Comment:
This function is almost entirely duplicated with the copy below, can you find aware to reduce the code duplication here?

Reply:
Defined a template function. Instantiated with `vector<string>` and `vector<vector<uint8_t>>`. Should resolve the code duplication issue. Could you check if this is ok/requires any changes?